### PR TITLE
Preset colorbar options

### PR DESCRIPTION
**Pull request by kevindougherty-noaa**
Addresses Issue #32. Preset values for specific variables and evaluation types. 

Noting that specific humidity values are catered more toward near surface. Winds (zonal and meridional) have attempted to be catered towards all levels with a max level of 50 (min level -50).

### CODE REVIEW

File: src/emcpy/plots/tools.py
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/tools.py]: vmax is -5?

File: src/emcpy/plots/tools.py
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/tools.py]: vmin=vmax?

File: src/emcpy/plots/tools.py
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/tools.py]: 200K is probably a bit extreme? that's -70C, but I guess I don't have a good grasp for 200hPa temps right now

File: src/emcpy/plots/tools.py
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/tools.py]: what about a default value to return here for everything? Or is the assumption that the user should not call this if they want default?

**Review by CoryMartin-NOAA — CHANGES_REQUESTED** [review]

File: src/emcpy/plots/tools.py
Comment (kevindougherty-noaa) [reply src/emcpy/plots/tools.py]: mixed up the vmin/vmax numbers for differences. Will fix.

File: src/emcpy/plots/tools.py
Comment (kevindougherty-noaa) [reply src/emcpy/plots/tools.py]: There is already default values set in the plotting classes. The assumption should be that this is only to be called if they want preset values. Regardless, you will have to set the plotting object's vmin, vmax, and colormap equal to these values i.e.
```
scatterobj = MapScatter(lat, lon, data)
cbar_presets = ColorbarPreset('temperature', 'diff')
scatterobj.vmin = cbar_presets.vmin
scatterobj.vmax = cbar_presets.vmax
scatterobj.cmap = cbar_presets.cmap
```

Comment (CoryMartin-NOAA) [reply src/emcpy/plots/tools.py]: Ah okay got it, so you would presumably do something like:
```
if var in ['temperature', 'brightness temperature']: # etcetera
    cbar_presets = ColorbarPreset(var, 'diff')
    scatterobj.vmin = cbar_presets.vmin
    scatterobj.vmax = cbar_presets.vmax
    scatterobj.cmap = cbar_presets.cmap
```

I guess I'm asking is this the best way to do it? I'm not sure.

File: src/emcpy/plots/tools.py
Comment (kevindougherty-noaa) [reply src/emcpy/plots/tools.py]: I can bump it to 220K so its roughly -50C. the colorbar can always be set to 'extend' to show higher values as well. That way its a consistent 100K spread

File: src/emcpy/plots/tools.py
Comment (aerorahul) [reply src/emcpy/plots/tools.py]: So I don't like how all this is defined inside of `__init__`.
The `__init__` method should provide defaults and everyone else should inherit from `ColorbarPreset` and overwrite for their specific variable.

**Review by aerorahul — CHANGES_REQUESTED** [review]

File: src/emcpy/plots/tools.py
Comment (CoryMartin-NOAA) [reply src/emcpy/plots/tools.py]: So you want a method to update the defaults based off of the variable?
Like this:
```
mycbar = ColorbarPreset()
mycbar.defaults('temperature', 'diff')
```
?

Comment (kevindougherty-noaa) [reply src/emcpy/plots/tools.py]: Maybe something like this?
```
scatterobj = Scatter(lat, lon, data)
ColorbarPreset(scatterobj, variable='temperature', eval_type='diff')
```
the constructor would then look like this:
```
def __init__(self, plotobj, variable, eval_type):
    self._get_cbar_type(variable, eval_type)

    plotobj.vmin = self.vmin
    plotobj.vmax = self.vmax
    plotobj.cbar = self.cbar
```
with `_get_cbar_type()` being a method with all of this information in it?

Comment (aerorahul) [reply src/emcpy/plots/tools.py]: why just the colorbar type?
Why not use this to get entire field specs.  min, max, contour interval, contour labels, units, etc.

Comment (aerorahul) [reply src/emcpy/plots/tools.py]: I am thinking something along these lines:

```
class VariableSpecs:
    def __init__(self, vardict=dict())
        self.variable_name = vardict.get('name', 'VariableName')
        self.variable_sname = vardict.get('short_name', 'VName')
        self.variable_type = vardict.get('type', 'VariableType')
        self.variable_units = vardict.get('units', 'VariableUnits')

        return

class FieldSpecs:
    def __init__(self, var_obj=VariableSpecs()):
        self.range = vardict.get('range', [-1., 1.])
        self.cntr_int = vardict.get('contour_interval', 0.1)
        self.cntrs = np.arange(self.range[0], self.range[-1], self.cntr_int)

        return

class ColorbarPreset:
    def __init__(self, colobardict=dict()):

        self.eval_type = colobardict.get('eval_type', 'diff')
        _cmap = colorbardict.get('cmap', None)
        if _cmap is None:
            self.cmap = _cmap
        else:
            self.cmap = 'coolwarm' if self.eval_type in ['diff'] else 'rainbow'

        return
```
and then extending this for variables eg. `Temperature`, `Sp. Humidity`, `Ice`, etc. and fields e.g. `T500`, `Q850`, etc.

Thoughts?

Comment (CoryMartin-NOAA) [reply src/emcpy/plots/tools.py]: So then the vardict is defined in another spot that is like this?:

```

rootvardict = {'temperature': {'range': [220, 320],
                                                  'contour_interval': 5,
                                                  'name': 'temperature',
                                                  'units': 'K',
                                                 }
                      }

def get_vardict(varname):
    return rootvardict[varname]

vardict = get_vardict('temperature')
```

I do agree we should probably try to make this more generic than just `colorbar`, for example, a time series of temperature might also want the y-axis range to be the same as the colorbar range

**Comment by CoryMartin-NOAA** [comment]
Tests fail. Is this because the other PR is not yet merged?

File: src/emcpy/plots/variable_specs.py
Comment (aerorahul) [inline src/emcpy/plots/variable_specs.py]: Can we put this in a yaml?
And load the yaml from a `cfg` directory?
Don't need to come here to make changes if someone wants to change colormap or range, etc.

Comment (CoryMartin-NOAA) [reply src/emcpy/plots/variable_specs.py]: I like this suggestion, but suggest that we do this at a later date (create an issue in the repo to add this functionality) as that will introduce a `yaml` dependency for `emcpy`

Comment (aerorahul) [reply src/emcpy/plots/variable_specs.py]: This is fine for now.

Comment (aerorahul) [reply src/emcpy/plots/variable_specs.py]: make an issue and mark it to do later.

File: src/emcpy/plots/variable_specs.py
Comment (aerorahul) [inline src/emcpy/plots/variable_specs.py]:
```suggestion
        'diff contour interval': 0.5,
```

**Review by aerorahul — CHANGES_REQUESTED** [review]

**Comment by kevindougherty-noaa** [comment]
> Tests fail. Is this because the other PR is not yet merged?

Yes. The `__init__.py` includes something from PR #29 so once that is approved, I will merge and the tests should pass.

**Review by aerorahul — APPROVED** [review]

**Review by CoryMartin-NOAA — APPROVED** [review]
Approved assuming you create an issue to use YAML later for these defaults

**Pull request merged**
